api/src/main/java/org/openmrs/module/xdssender/api/xds/XdsUtil.java

```diff
@@ -69,12 +69,41 @@ public String parseCcdToHtml(Bundle resource, File ccdTemplate) throws IOExcepti
 //                    TODO - Rework all the mappings from Observations- limiting the fetch to height,weight,temp,pulse,BP(both systolic and diastolic)
                     Observation obs = (Observation) eResource;
 //                    System.out.println("Processing Obs:====> "+obs.getCode().getCodingFirstRep().getDisplay());
+
                     if (codes.contains(obs.getCode().getCodingFirstRep().getCode())) {
                         vitalSigns.add(mapObservationResource(obs));
-                    }else if(obs.getCode().getCodingFirstRep().getCode().equals("984AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA")){
-//                        Process the immunizations,
-                        Immunization medication = mapImmunizationResource(obs);
-                        immunizations.add(medication);
+                    } else {
+//                        Process other obs
+                        switch (obs.getCode().getCodingFirstRep().getCode()) {
+                            case "984AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA": {
+//                                Process the immunizations
+                                Immunization medication = mapImmunizationResource(obs);
+                                immunizations.add(medication);
+                                break;
+                            }
+                            case "1284AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA":
+                            case "159614AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA": {
+//                                Process tonsillitis diagnosis
+                                Condition condition = mapConditionResource(obs);
+                                conditions.add(condition);
+                                break;
+                            }
+//                            case "1442AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"://Nevirapine{
+                            case "1282AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"://AMOXICILLIN
```

Comment:
The code for Current Medication Order is `1442AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA` I think, no?